src/api/cvc4cpp.h

```diff
@@ -0,0 +1,4556 @@
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   * Turned into a conjunction of binary applications of the operator on
+   * adjoining parameters.
+   */
+  CHAIN,
+  /**
+   * Chained operator.
+   * Parameters: 1
+   *   -[1]: the kind of the binary operation.
+   * Create with:
+   *   mkOpTerm(Kind opkind, Kind kind)
+   */
+  CHAIN_OP,
+
+  /* Boolean --------------------------------------------------------------- */
+
+  /**
+   * Boolean constant.
+   * Parameters: 1
+   *   -[1]: the Boolean value of the constant
+   * Create with:
+   *   mkTrue()
+   *   mkFalse()
+   *   mkBoolean(bool val)
+   *   mkConst(Kind kind, bool arg)
+   */
+  CONST_BOOLEAN,
+  /* Logical not.
+   * Parameters: 1
+   *   -[1]: the Boolean Term to negate
+   * Create with:
+   *   mkTerm(Kind kind, Term child)
+   */
+  NOT,
+  /* Logical and.
+   * Parameters: n > 1
+   *   -[1]..[n]: the Boolean Term of the conjunction
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  AND,
+  /**
+   * Logical implication.
+   * Parameters: 2
+   *   -[1]..[2]: Boolean Terms, [1] implies [2]
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  IMPLIES,
+  /* Logical or.
+   * Parameters: n > 1
+   *   -[1]..[n]: the Boolean Term of the disjunction
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  OR,
+  /* Logical exclusive or.
+   * Parameters: 2
+   *   -[1]..[2]: Boolean Terms, [1] xor [2]
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  XOR,
+  /* If-then-else.
+   * Parameters: 3
+   *   -[1] is a Boolean condition Term
+   *   -[2] the 'then' Term
+   *   -[3] the 'else' Term
+   * 'then' and 'else' term must have same base sort.
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  ITE,
+
+  /* UF -------------------------------------------------------------------- */
+
+  /* Application of an uninterpreted function.
+   * Parameters: n > 1
+   *   -[1]: the function Term
+   *   -[2]..[n]: function argument instantiation Terms
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  APPLY_UF,
+  /**
+   * Boolean term variable
+   * ???
+   */
+  BOOLEAN_TERM_VARIABLE,
```

Comment:
@ajreynol - same question as above.  User's shouldn't ever create BOOLEAN_TERM_VARIABLE's, right?  Can they be returned as a result of getValue?  I don't think so, right?

Reply:
Yes, these are internal.  They cannot be returned as a result of getValue.